src/plugins/shared/clusterOptions/options/selectExistingClusterOption.ts

```diff
@@ -0,0 +1,77 @@
+import * as vscode from "vscode";
+import { ReadyAzureSessionProvider } from "../../../../auth/types";
+import { ClusterPreference } from "../../types";
+import { selectSubscriptions } from "../../../../commands/aksAccount/aksAccount";
+import { getClusters, getManagedCluster, getKubeconfigYaml, Cluster } from "../../../../commands/utils/clusters";
+import { getFilteredSubscriptions } from "../../../../commands/utils/config";
+import { Errorable, failed } from "../../../../commands/utils/errorable";
+import { longRunning } from "../../../../commands/utils/host";
+import { RecentCluster } from "../state/recentCluster";
+import { handleNoSubscriptionsFound } from "../../../../commands/utils/subscriptions";
+
+export async function selectExistingClusterOption(
+    sessionProvider: ReadyAzureSessionProvider,
+): Promise<Errorable<ClusterPreference>> {
+    await selectSubscriptions();
+
+    const subscriptions = getFilteredSubscriptions();
+    if(subscriptions.length === 0) { 
+        handleNoSubscriptionsFound();
+        return { succeeded: false, error: "No subscriptions found." };
+    }
+
+    const clusters = (
```

Comment:
suggestion here to add: I hit this last time when I had a subscription without any aks clusters

```
cluster.length === 0 and show a error message ?

```

Reply:
Ah yes, good one, will include error handling for no clusters in follow up PR